swift/ios-shoppe-demo/ServiceManagers/AppStateController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  AppStateController.swift
+//  ios-shoppe-demo
+//
+//  Created by Harold Davis Jr. on 3/30/20.
+//  Copyright © 2020 Harold Davis Jr. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class AppStateController {
```

Comment:
As we add more features to the app, let's begin adding comments on design and behavior.  The AppStateController seems like it's doing 2 things: 1) it's storing user information (i.e. app state like the class name suggests) but 2) it's also controlling UI.  Normally, I'd separate these as two concerns: one controller should handle data and the other could be responsible for UI.

The `displayAlert` function seems like a convenience method to create an login form.  If we wanted a factory to create generic alerts, you could have a `static func displayAlert` that simply sets up the `UIAlertController`.  The username could then be obtained within the `okAction` with

```swift
let username = createUserAlert.textFields?[0].text
// somecontroller.setState('username', username)
```

In this manner, there's no need to store a reference to the textfield (it already exists within the state of the Alert).  This could also be refactored to a `UserLogin` standalone artifact to provide better semantics of its design.

WDYT?

Reply:
I think your right I'm starting to debate the needs in this "state controller"